lib/plenario2/actions/meta_actions.ex

```diff
@@ -62,8 +67,75 @@ defmodule Plenario2.Actions.MetaActions do
     |> Repo.insert()
   end
 
-  ##
-  # update
+  @doc """
+  Load a single row of `Meta` with any field specified by `preloads`
+  preloaded.
+
+  ## Examples
+
+    iex> get_by_pk_preload(1)
+    iex> get_by_pk_preload(1, [:data_set_constraints])
+    iex> get_by_pk_preload(1, [:data_set_constraints, :data_set_diffs])
+
+  """
+  @spec get_by_pk_preload(pk :: integer, preloads :: list) :: Meta
+  def get_by_pk_preload(pk, preloads \\ []) do
+    Repo.one(
+      from(
+        m in Meta,
+        where: m.id == ^pk,
+        preload: ^preloads
+      )
+    )
+  end
+
+  @doc """
+  Get a list of column names for a `Meta` struct. 
+
+  ## Examples
+
+    iex> get_columns(meta)
+    ["id", "location", "datetime", "observation"]
+
+  """
+  @spec get_columns(meta :: Meta) :: list[charlist]
+  def get_columns(meta) do
+    meta = Repo.preload(meta, :data_set_fields)
+    for field <- meta.data_set_fields() do
+      field.name()
```

Comment:
Per my general comment, I hate the `field.name()` instead of just a plain `field.name`.